azure-pipelines.yml

```diff
@@ -558,12 +558,16 @@ stages:
       env:
         SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.6-bin-hadoop2.7
 
+    # Spark 3.0.0 uses Arrow 0.15.1, which contains a new Arrow spec. This breaks backward
```

Comment:
At the moment there are no published workers that's backward compatible with 3.0 (since the previous workers only use 0.14.1 and aren't aware of the new spec).  But I agree that this is a breaking change.

For backward compatibility, do we want to differentiate between different spark versions and test them against different spark Worker versions?  Or one Worker version where we say is backward compatible for all spark versions?

This can be addressed in a separate PR if needed.

Reply:
> For backward compatibility, do we want to differentiate between different spark versions and test them against different spark Worker versions? Or one Worker version where we say is backward compatible for all spark versions?

I would say the latter.

Reply:
Then I think we will have to wait until the next official Worker release before we can remove these extra filters.

Reply:
... since we have one worker binary to support all spark versions.

Reply:
Well, you remove the backward compatibility test (breaking change) then added it back when the new one is released.

Reply:
Do you want me to remove the extra filters for 3.0 and in the unit tests add the skip attribute ?  Or just remove the spark 3.0.0 section in the backward compatibility tests.

Reply:
Removed 3.0 backward compatibility testing.